### DISCUSSION TITLE
NameGenerator not working with shared project

### DISCUSSION THREAD

**Discussion by Ruttie2006** [AvaloniaUI/Avalonia#9384] [Q&A] [answered]
I made a `AppName.Shared` project, and a `AppName.Windows` head, but the NameGenerator analyzer does not appear to work on the files of the shared project. Is there any way to make it work?

**Comment by Ruttie2006** [comment]
Of course, the easiest way to fix it is by not using a shared project, but I'd rather get it to work with a shared project.

**Comment by maxkatz6** [comment, marked as answer]
I don't think it's possible to run source generators on a shared project. Why would you use a deprecated projects model?

**Reply by Ruttie2006** [reply]
Well, there are things I consider to be advantages of using a shared project, mainly the ease of adding platform-specific code and assets.
